examples/orders_create_and_download_multiple_orders.py

```diff
@@ -75,8 +75,8 @@ async def create_and_download(order_detail, directory, client):
         order = await client.create_order(order_detail)
         reporter.update(state='created', order_id=order.id)
 
-        # poll
-        await client.poll(order.id, report=reporter.update)
+        # wait for completion
+        await client.wait(order.id, report=reporter.update_state)
```

Comment:
@jreiberkyle what would you think about wait taking an instance of Order (like the one from line 75) instead of a string id? I think that could help make the API more consistent and intuitive?

Also, I'd like to understand the reporter feature better. What is it about?